### PR TITLE
feat: make `IsOptional` conditional

### PR DESCRIPTION
**Pull request by yitzchak-ben-ezra-ecoplant**
## Description

**There is already an open approved PR which is not merged!
This is a duplicate just to get this merged already**
https://github.com/typestack/class-validator/pull/196
AND
https://github.com/typestack/class-validator/issues/651

## Checklist

- [x] the pull request title describes what this PR does (not a vague title like `Update index.md`)
- [x] the pull request targets the *default* branch of the repository (`develop`)
- [x] the code follows the established code style of the repository
  -[x] `npm run prettier:check` passes
  -[x] `npm run lint:check` passes
- [x] tests are added for the changes I made (if any source code was modified)
- [x] documentation added or updated
- [x] I have run the project locally and verified that there are no errors

### Fixes

fixes #[issue number], fixes #[issue number]

### CODE REVIEW

File: src/decorator/common/IsOptional.ts [hunk 1]
```diff
@@ -4,22 +4,37 @@ import { ValidationTypes } from '../../validation/ValidationTypes';
...
+export function IsOptional(
+  condition?: IsOptionalCondition | ValidationOptions,
+  validationOptions?: ValidationOptions
+): PropertyDecorator {
   return function (object: object, propertyName: string): void {
     const args: ValidationMetadataArgs = {
       type: ValidationTypes.CONDITIONAL_VALIDATION,
       target: object.constructor,
-      propertyName: propertyName,
+      propertyName,
```
Comment (erabl-identos) [inline src/decorator/common/IsOptional.ts:23]:
```suggestion
      propertyName: propertyName,
```

**Comment by yitzchak-ben-ezra-ecoplant** [comment]
@erabl-identos please complete the review so it gets merged...

File: test/functional/conditional-validation.spec.ts [hunk 2]
```diff
@@ -92,4 +92,72 @@ describe('conditional validation', () => {
...
+    return validator.validate(model).then(errors => {
+      expect(errors.length).toEqual(1);
+      expect(errors[0].target).toEqual(model);
+      expect(errors[0].property).toEqual('title');
+      expect(errors[0].constraints).toEqual({ equals: 'title must be equal to test' });
+      expect(errors[0].value).toEqual('bad_value');
+    });
+  });
+
+  it('should validate a property when condition returns true and value is not supplied', function () {
```
Comment (erabl-identos) [inline test/functional/conditional-validation.spec.ts:132]: when the condition returns true (meaning the property is optional), the validation should _not_ happen if the property was not supplied.

**Review by erabl-identos — COMMENTED** [review]
looking forward to use this feature

**Review by erabl-identos — DISMISSED** [review]

**Comment by yitzchak-ben-ezra-ecoplant** [comment]
@erabl-identos ready for approval

**Review by erabl-identos — DISMISSED** [review]

**Comment by erabl-identos** [comment]
According to Github at least one approval must be provided by somebody with write access.

Maybe @NoNameProvided can move this forward?

**Comment by yitzchak-ben-ezra-ecoplant** [comment]
@NoNameProvided Please close this.. we really need this.. and it's waiting for more than 2 years...

**Comment by yitzchak-ben-ezra-ecoplant** [comment]
Maybe @pleerock / @vlapo - Do you have write access and can review this minor PR?

**Comment by ybeapps** [comment]
Looks like this repo is dead. No maintainers comments in this minor PR since it’s been created.. almost 2 weeks..
Only the bot doing any work here..
Too bad.. it’s a good repo, let others who have more time to maintain it for the world..

**Comment by yitzchak-ben-ezra-ecoplant** [comment]
Closing this PR since this repo is DEAD!!
I've forked this repo.. and will open it for public edits.. whoever wants to help maintain it - your welcome!
https://github.com/yitzchak-ben-ezra-ecoplant/class-validator-forked

**Comment by NoNameProvided** [comment]
https://github.com/typestack/class-validator/pull/932

**Pull request closed without merging**